src/traces/bar/style.js

```diff
@@ -71,6 +71,8 @@ function styleOnSelect(gd, cd) {
         stylePointsInSelectionMode(s, trace, gd);
     } else {
         stylePoints(s, trace, gd);
+
+        Registry.getComponentMethod('errorbars', 'style')(s);
```

Comment:
Hmm. I'm not sure this is the correct fix. Look at https://codepen.io/etpinard/pen/gEKqgj?editors=1010 (the same as your _before_ example, but with scatter traces). Selection doesn't dim the error bars.

Reply:
Back in 1.36.0, selecting bars didn't dim the associated error bars: https://codepen.io/etpinard/pen/jJKdYN?editors=1000

We should probably eventually add a way to dim the error bars on selections (by adding a `[un]selected.error_(x|y).color` or by adding a trace-wide `[un]selected.opacity`), but that's for another day.

To _fix_ this bug, I think we should make sure error bars on bar traces don't get dimmed on selection by default.

Reply:
The bar version is different from `scatter` possibly because in `bar` the select deals with both bar points and errorbar points. So I still think it may be a good patch to solve the regression.

Reply:
.... and why should scatter and bar error bars behave differently?

Reply:
... but yeah you're right that the regression is way worse than the inconsistency this fix brings up.

Reply:
> .... and why should scatter and bar error bars behave differently?

In `scatter` within `stylePoint` function `selectAll('path.point')` is applied; whereas in `bar` within `stylePoint` function `selectAll('path')` is applied which resulted in errorbars being selected after click.